.github/workflows/tiiuae-pixhawk-and-saluki.yaml

```diff
@@ -9,276 +9,33 @@ on:
   workflow_dispatch:
 
 jobs:
-  pixhawk:
-    name: build pixhawk
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout px4-firmware
-        uses: actions/checkout@v3
-        with:
-          path: px4-firmware
-          fetch-depth: 0
-      - name: Run px4-firmware pixhawk build
-        run: |
-          set -eux
-          mkdir -p bin
-          cd px4-firmware/
-          ./clone_public.sh
```

Comment:
here is ran clone_public.sh but in other steps we don't run this. In other steps we are cloning submodules instead. Maybe there is a reason for this, so I have kept it this way with reusable workflow as well. Controlled by clone_public flag

Reply:
clone_public skips all private submodules, so this action for pixhawk can be run also in forked repos.